### PR TITLE
MOAR WINDOZE CLEANUP

### PR DESCRIPTION
**Pull request by saghul**
Yak shaving at it's best.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit-windows/44/
R=@libuv/collaborators

### CODE REVIEW

File: src/win/util.c [hunk 1]
```diff
@@ -1299,20 +1242,34 @@ void uv_os_free_passwd(uv_passwd_t* pwd) {
...
+                                0,
+                                NULL,
+                                NULL);
 
   if (bufsize == 0)
     return uv_translate_sys_error(GetLastError());
 
   /* Allocate the destination buffer */
-  *utf8 = uv__malloc(bufsize);
+  *utf8 = uv__malloc(bufsize + 1);
```
Comment (cjihrig) [inline src/win/util.c]: I don't think the `+ 1` is required in all cases. It's probably simpler and safer to just always allocate it though.

Comment (saghul) [inline src/win/util.c]: yep, when the size is -1 is not really needed, but I thought I'd simplify.

**Comment by cjihrig** [comment]
LGTM

**Comment by txdv** [comment]
Amazing, though, could you add that exact explanation as a comment above the "+1"?

**Comment by saghul** [comment]
Landed in 4c945f49365ab4d6e1b07bf0ef2893455dc04622, 46d84eae28bbf82eb8e1ace35cead0bd73163a8e and e397caa3a6a139fff2fed06e0223ee92582fa2d2.

@txdv I added a note.

**Pull request merged**